File: clients/php/src/Result/RegisterResult.php
<?php

declare(strict_types=1);

namespace Priostack\Acn\Result;

/**
 * Result of {@see \Priostack\Acn\Client::register()}.
 *
 * The {@see $token} is shown once by the server; persist it to reconnect the
 * same agent later without registering again.
 */
final class RegisterResult
{
    /**
     * @param array<string, mixed> $raw The full, unwrapped {@code data} payload.
     */
    public function __construct(
        public readonly string $token,
        public readonly string $agentId,
        public readonly string $accountId,
        public readonly string $tier,
        public readonly array $raw = [],
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromData(array $data): self
    {
        return new self(
            token: (string) ($data['token'] ?? ''),
            agentId: (string) ($data['agentId'] ?? ''),
            accountId: (string) ($data['accountId'] ?? ''),
            tier: (string) ($data['tier'] ?? ''),
            raw: $data,
        );
    }
}

File: clients/java/src/main/java/com/priostack/acn/RegisterResult.java
package com.priostack.acn;

import com.fasterxml.jackson.databind.JsonNode;

/** Result of {@link ACNClient#register}. */
public final class RegisterResult {

    private final String token;
    private final String agentId;
    private final String accountId;
    private final String tier;
    private final JsonNode raw;

    public RegisterResult(
            String token, String agentId, String accountId, String tier, JsonNode raw) {
        this.token = token;
        this.agentId = agentId;
        this.accountId = accountId;
        this.tier = tier;
        this.raw = raw;
    }

    /** The bearer token, shown once. Persist it to reconnect this agent later. */
    public String token() {
        return token;
    }

    public String agentId() {
        return agentId;
    }

    public String accountId() {
        return accountId;
    }

    public String tier() {
        return tier;
    }

    /** The full raw {@code data} object the server returned. */
    public JsonNode raw() {
        return raw;
    }

    @Override
    public String toString() {
        return "RegisterResult{agentId=" + agentId + ", accountId=" + accountId + ", tier=" + tier + '}';
    }
}
